ext/stack_frames/buffer.c

```diff
@@ -11,12 +11,17 @@ typedef struct {
 static void buffer_mark(void *ptr)
 {
     buffer_t *buffer = ptr;
+    VALUE frame;
 
     for (int i = 0; i < buffer->length; i++) {
-        rb_gc_mark(buffer->profile_frames[i]);
+        if (RTEST(frame = buffer->profile_frames[i])) {
+            rb_gc_mark(frame);
+        }
     }
     for (int i = 0; i < buffer->capacity; i++) {
-        rb_gc_mark(buffer->frames[i]);
+        if (RTEST(frame = buffer->frames[i])) {
```

Comment:
Looking at the code in `buffer_initialize`, shouldn't `buffer->frames[i]` never be null from `0 <= i < buffer->capacity`?

Reply:
Yep, as said in the PR description, it's redundant, but I wanted to be extra sure :p 

But yeah I could drop that extra check. I'm not even sure if it's a problem to `rb_gc_mark(Qnil)`.